src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -85,7 +85,7 @@ private void RunJTFWithCorrectContext(Project project, Func<Task> asyncTask)
                 });
             }
 
-            PumpingJTF.Run(asyncTask);
+            PumpingJTF.RunAsync(asyncTask);
```

Comment:
Calling `RunAsync` instead of `Run`, without doing anything with the result, is an almost certain way to remove a deadlock. But it generally doesn't fix the underlying problem that caused `Run` to deadlock in the first place -- it just sweeps it under the rug. 
I'm also concerned about the significant behavioral change that `RunJTFWithCorrectContext` used to return only after invoking the async delegate but now it returns after just kicking it off. That's *very* significant, and its impact can't be assessed simply by looking at this changed line. Are all callers of this method OK with `asyncTask` not completing before this method returns? If so, this is a good change (even if it doesn't resolve the underlying deadlock) since it means the calling thread will not have to synchronously block.

We should be sure we understand the underlying deadlock though, or else this could just kick the deadlock down the road to VS shutdown, where it may still deadlock (assuming `PumpingJTF` feeds async work into a `JoinableTaskCollection` that blocks on shutdown, as it should).